### ISSUE TITLE
Object Error

### ISSUE THREAD

**Issue by minhqnh99** [LINKIWI/brick-pop-solver#1]
So basically i got this issue:

Reading board image...
Traceback (most recent call last):
  File "src/solve.py", line 222, in <module>
    main()
  File "src/solve.py", line 218, in main
    return solve(sys.argv[1])
  File "src/solve.py", line 183, in solve
    board = load_board(board_image_file_name)
  File "src/solve.py", line 151, in load_board
    bgr = img[pixel_i][pixel_j]
TypeError: 'NoneType' object has no attribute '__getitem__'

And also i have a 1080x1920 screen resolution, how do i edit the offset ?

**Comment by LINKIWI** [comment]
Are you sure you passed in the file name properly as a command-line option? `img` is `None` if OpenCV errors in reading the file.

> And also i have a 1080x1920 screen resolution, how do i edit the offset ?

Unfortunately you will have to adjust it manually by modifying the constants `IMAGE_BLOCK_OFFSET`, `IMAGE_BLOCK_START_I`, and `IMAGE_BLOCK_START_J` at the top of the file.

**Comment by minhqnh99** [comment]
You mean the $ ./brick-pop-solve.sh ?

About the resolution, i know i have to adjust it manually but i mean how can i calculate what value to fill in ?

**Comment by LINKIWI** [comment]
> You mean the $ ./brick-pop-solve.sh ?

Yes, can you paste the output of running that script? Seems like ADB isn't pulling the screenshot properly.

> About the resolution, i know i have to adjust it manually but i mean how can i calculate what value to fill in ?

Use a photo editor like GIMP to check the pixel offsets. Unfortunately it's a very manual process.

**Comment by minhqnh99** [comment]
Here's the output:

> $ ./brick-pop-solve.sh
List of devices attached
HSMDU17523000099        device

Reading board image...
Traceback (most recent call last):
  File "src/solve.py", line 222, in <module>
    main()
  File "src/solve.py", line 218, in main
    return solve(sys.argv[1])
  File "src/solve.py", line 183, in solve
    board = load_board(board_image_file_name)
  File "src/solve.py", line 151, in load_board
    bgr = img[pixel_i][pixel_j]
TypeError: 'NoneType' object has no attribute '__getitem__'

**Comment by LINKIWI** [comment]
My best guess is

```bash
$ adb shell screencap -p | sed 's/\r$//' > brick-pop.png
```

isn't producing the screenshot correctly. Try running that manually and see if it pulls a screenshot.

Or, there's an error in reading the image via OpenCV. What version do you have installed?

**Comment by minhqnh99** [comment]
I installed Python 2.7. Also i just ran that code and it produced a brick-pop.png but said can't open the image

**Comment by LINKIWI** [comment]
Then it's probably a problem with your OpenCV installation. Try a reinstalling with the latest SHA on https://github.com/opencv/opencv

**Comment by minhqnh99** [comment]
I just tried a newer OpenCV and now i got this:

Reading board image...
Traceback (most recent call last):
  File "src/solve.py", line 222, in <module>
    main()
  File "src/solve.py", line 218, in main
    return solve(sys.argv[1])
  File "src/solve.py", line 183, in solve
    board = load_board(board_image_file_name)
  File "src/solve.py", line 144, in load_board
    img = cv2.imread(board_image_file_name, cv2.IMREAD_COLOR)
AttributeError: 'module' object has no attribute 'IMREAD_COLOR'

**Comment by LINKIWI** [comment]
What OpenCV SHA are you on? What is `cv2.__version__`?

I just tried myself and as far as I can tell it works on SHA `fee2049642cacc1ff9a2d4f7bf8ef54ca1759981` of OpenCV.
